File: src/myAnswers/ValidPalindrome.java
package myAnswers;

public class ValidPalindrome {
	
	public boolean isPalindrome(String s) {
		
		// remove all characters that are non-alphanumeric.
		String newS = s.replaceAll("[^A-Za-z0-9]", "");
			
		// create a new String that is reverse to the original input
		String reverse = new StringBuilder(newS).reverse().toString();
		
		return newS.equalsIgnoreCase(reverse);
	}

}

File: src/myAnswers/ValidPalindrome.py
# The solution code of 'Valid Palindrome'

def isPalindrome(s):
    '''
    Input: a String
    Output: a boolean value

    Check whether a given string is a palindrome or not.
    
    '''
    # create a new string without non-alphanumeric characters.
    newS = ''.join([char for char in s if char.isalnum()]).lower()

    # check whether it is a valid palindrome
    return (newS == newS[::-1])
    
